classes/core/db/QueryPager.php

```
@@ -37,4 +53,43 @@ public function getQuery() {
         $newSql .= " limit " . $this->paging->getFirstRecord() . ", " . $this->paging->getRecordsPerPage();
         return $newSql;
     }
+    
+    /**
+     * Returns the query, with the filtering applied to it.
+     * Use the SQLBuilder to add a where condition.
+     * Given a list of column names and search string, populate
+     * 1. $condition - SQL clause
+     * 2. $vartypes - types of parameters and
+     * 3. $args - list of search strings
+     * to be used for prepared statements.
+     *
+     * Example:
+     *  sqlBuilder.where('name like ? or age like ?', 'ss', 'john', 'john');
+     *
+     * @see SQLBuilder::filter
+     * @return String the sql query
+     */
+    private function applySearchFilter($search) {
+     
+        $searchString = $search->getSearchString();
+        $searchColumns = $search->getSearchColumns();
+        $numColumns = count($searchColumns);
+         
+        if (empty($searchString) || $numColumns < 1) return;
+         
+        if ($this->sql instanceof SQLBuilder) {
+            $conditions = array();
+            foreach ($searchColumns as $column) {
+                $conditions[] = $column . " like ?";
+            }
+             
+            $varTypes = str_repeat('s', $numColumns); // types of parameters passed
+            $args = array_fill(0, $numColumns, "%$searchString%"); // list of search strings repeated up to number of columns
+             
+            call_user_func_array(array($this->sql, "filter"), array_merge(array(implode(" or ", $conditions), $varTypes), $args));
+           
+        } else {
+            Logger::warning("Does not support plain sql: Use SQLBuilder. Called by: QueryPager");

```

This may happen in case of a programming error, right?
I would throw an exception here (IllegalStateException),
In which case, do it above. If $this->sql is not instance of... then throw the exception.